### PR TITLE
Feature 495: Changed pictogram box changes

### LINKED ISSUE #495: Pictogram text is cut off instead of being divided over two lines in portrait mode on 600x1024 screen
**Describe the bug**
When a pictogram's text doesn't fit on one line it is cut off so some of it doesn't show.

**To Reproduce**
1. Go to a weekplan.
2. Make sure pictogram text is enabled under settings.
3. If an activity with the bug doesn't already exist, add an activity with the pictogram in the screenshot (search "se udskrift").
4. Observe bug.

**Expected behavior**
I would expect the pictogram text to span two lines or resize to fit.

**Actual behavior**
Part of the pictogram text simply wasn't shown.

**Screenshots**
Pictogram text as seen on the weekplan:
image
Full pictogram text:
image

**Environment (please complete the following information):**
Observed on emulator '7 WSVGA (Tablet) API 29'. Resolution: 600x1024

### PR DESCRIPTION
**Pull request by Riasalit**
fixes #495 
The Issue was that some text was cut off when it spanned multiple lines, as the box size was static. this has been solved by allowing the pictogram box to be as high as the elements within and set the max amount of lines for the text to 2.

this made smaller words split into 2 lines, which was further solved by only allowing the text to be 1 line of only 1 word was present in the string.

### CODE REVIEW

**Review by AdrianPlesner — CHANGES_REQUESTED** [review]
**Code Design**
​
- [x] The code is in the right place? (Both in terms of folder structure and class structure)
- [x] This code could not, to the best of my knowledge, have reused existing code
- [x] The code is not over-engineered. Examples of over-engineering: Implemented behavior for future problems
- [x] The code does not introduce functionality that is necessary for solving the problem
​
**Code Readability and Maintainability**
​
- [x] Names are meaningful and self-documenting
- [x] It is understandable, by reading the code, what it does
- [x] The code is simple and readable. i.e. it contains no "hack", or obscure solution
- [ ] All new functionality is covered by tests
- [ ] It is understandable, by reading the tests, what they do
- [ ] The tests covers sensible cases. Both happy paths and exception paths
- [ ] The tests cover the full functionality. i.e., The tests fails if some of the requested functionality is missing
- [x] Error messages in the code are understandable
- [x] Confusing sections of code are covered by comments
​
**Functionality**
​
For this part, open the app, and test:
​
- [ ] The features that the code claims to implement, are actually implemented
- [x] The code does not look like it has bugs
- [x] All the new screens are reachable through in-app navigation
​
**Points for Discussion and Reflection**
​
- [x] UI messages, that the user might meet, are actually helpful
- [x] There are no errors that are bad practice. E.g. hard-coding a test database connection
- [x] There is well written code in this. Remember to tell the developer!

Hvis der er ét langt ord bliver den stadig ikke delt op, husk at tage højde for at bredden kan være forskellig på forskellige enheder. 
Desuden mangler der test.

**Comment by Riasalit** [comment]
Ved virkelig ikke hvordan man skal kunne tæste at størelsen på en tekstbox vokser nedad, når et ord er for langt til en linje når det kommer an på størrelsen af skærmen, boksen, teksten. altså alt der er lavet er at max lines i elementet der auto justere det hele, er ændret til at være slightly dynamisk og så kan tekstboxen vokse

File: lib/widgets/pictogram_text.dart [hunk 1]
```diff
@@ -83,3 +83,34 @@ class PictogramText extends StatelessWidget {
...
+          textScaleFactor: MediaQuery.of(context).textScaleFactor,
+          textDirection: TextDirection.ltr)
+        ..layout())
+      .size
+      .width;
+}
+
+/// Counts the amount of words by splitting by the spaces and count the result
+/// if theres only 1 word then max lines is 1 else the max lines is 2
+int textLines(String pictogramText, BuildContext context) {
```
Comment (AdrianPlesner) [inline lib/widgets/pictogram_text.dart:105]: Jeg tænker godt at denne funktion kan testes. 🙂

Comment (Riasalit) [reply lib/widgets/pictogram_text.dart:105]: Problemet er bare at alle test i programmet lige nu er widget tests, det vil sige at der så skal laves det widget hvor det kode ligger i, så jeg kan muligvis godt sige at den har en bestemt størelse og så give det en lang nok test til at det bliver splittet. men expect metoden bruger så en fidner der kan lede efter elementer enten efter en tekst, et tag, eller en type. og de vil alle retunere true uanset om der er 2 eller 1 linje da teksten er der, uanset om det kan ses eller ej

**Review by AdrianPlesner — APPROVED** [review]

**Review by Emil7370 — APPROVED** [review]

**Comment by LivHolm** [comment]
Please use the following checklist @Emil7370 . Please check it off as you go along.
​
**Code Design**

- [ ] The code is in the right place? (Both in terms of folder structure and class structure)
- [ ] This code could not, to the best of my knowledge, have reused existing code
- [ ] The code is not over-engineered. Examples of over-engineering: Implemented behavior for future problems
- [ ] The code does not introduce functionality that is unnecessary for solving the problem

**Code Readability and Maintainability**

- [ ] Has all the added code been properly commented

Only check one of the items below:

- [ ] The functionality does **not** need be covered by tests
- [ ] The functionality needs to be covered by tests

Only use if the functionality needs to be covered by tests:

- [ ] All functionality is covered by tests
- [ ] The tests has sensible names
- [ ] By reading the tests you know what they cover        
- [ ] The tests cover sensible cases. Both happy paths and exception paths
- [ ] The tests cover the full functionality. i.e., The tests fails if some of the requested functionality is missing  

**Functionality**

For this part, open the app, and test:

- [ ] The features that the code claims to implement, are actually implemented
- [ ] The code does not look like it has bugs
- [ ] All the new screens are reachable through in-app navigation

**Pull request merged**